### ISSUE TITLE
Invariant Violation: Native module cannot be null.

### ISSUE THREAD

**Issue by jesuscc1993** [juanamd/react-native-background-timer-android#2]
While the module works in Android, it throws the following exception when the app runs in iOS:
`Invariant Violation: Native module cannot be null.`

I am aware the module is android only and I'm not using it on iOS but simply importing it like this
`import AndroidBackgroundTimer from 'react-native-background-timer-android';`
...results in this error.

I tried the following and it does prevent the error on iOS but it does not properly import on Android:
```
const AndroidBackgroundTimer = Platform.OS === 'android'
  ? require('react-native-background-timer-android')
  : undefined;
```

**Comment by juanamd** [comment]
Thanks for the PR. Merged.

**Comment by juanamd** [comment]
So, I realized after upgrading to the new version that eslint didn't recognize the module anymore since the "main" entry point from package.json was removed.
I added it again but pointing to index.android.js since index.js no longer exists (version 1.0.2). I don't have any iOS device or simulator to test if this doesn't break the previous fix.
Hopefully it won't, but otherwise we will need to figure out another way to do it.

**Comment by jesuscc1993** [comment]
Just tested your latest code and it added the issue back. Looks like setting up the main property makes the file load on both platforms instead of the specific implementation.

**Update:** I tried renaming the files and importing them based on the platform but it won't work either; it keeps saying the module cannot be null:

***index.js***
```
import { Platform } from 'react-native';

import AndroidBackgroundTimer from './android.js';
import IosBackgroundTimer from './ios.js';

export default (Platform.OS === 'android') ? AndroidBackgroundTimer : IosBackgroundTimer;
```

**Comment by jesuscc1993** [comment]
I found a flawed fix.

If I rename the files like this:
`index.android.js > timer.android.js`
`index.ios.js > timer.ios.js`

And then add this file as main:
```
import BackgroundTimer from 'react-native-background-timer-android/timer';

export default BackgroundTimer;
```

...it "works". The flaw is, as index.js doesn't actually exist, the IDE will not pick up what BackgroundTimer is in this particular file.

Does this workaround work for you, @juanamd? If so, I can open a new PR for this.

**Comment by juanamd** [comment]
Sorry for the super late reply. I must have been busy at the time and completely forgot about this.
I've just released version 1.0.4 which should address this issue. Again, I don't have any iOS devices so can't confirm for sure.
Basically, the JS side now checks if the native module is null (which should be when running on iOS) and doesn't set listeners or call any other methods if this is the case.
The only change when using the module now is that it will throw an error if you try to call any method from an iOS device (instead of failing silently with the dummy index.ios).
